st2common/st2common/util/keyvalue.py

```diff
@@ -78,13 +79,14 @@ def get_key(key=None, user=None, scope=None, decrypt=False):
     """
     if not isinstance(key, six.string_types):
         raise TypeError('Given key is not typeof string.')
+
     if not isinstance(decrypt, bool):
         raise TypeError('Decrypt parameter is not typeof bool.')
 
     if not user:
         user = UserDB(cfg.CONF.system_user.user)
 
-    scope, key_id = _derive_scope_and_key(key, user, scope)
+    scope, key_id = _derive_scope_and_key(key, user.name, scope)
```

Comment:
So it looks like we changed function signature - it looks like previously user was expected to be username string and not the actual `UserDB` object.

Would be good to document that via docstring and perhaps also add an assert guard.

Reply:
Actually, looking at the code more, it seems like there is no need for the whole `UserDB` object so I would just prefer to pass in `user.name` instead of the whole user object.

Reply:
And if we decide to stick with `UserDB` object, I would prefer to call the argument `user_db` so it's consistent with other places and makes it immediately obvious what kind of argument it accepts.

Reply:
I just had a look at the Slack discussion again and it looks like some other functions require actual `UserDB` object.

If that's the case, I'm fine with that change, I would just prefer to rename argument from `user` to `user_db` to make it immediately clear and obvious (other function also takes `user_db` argument).

Reply:
@Kami The get_key function is kind of a mess.
1. We see user being set to UserDB by default if user not pass at https://github.com/StackStorm/st2/blob/master/st2common/st2common/util/keyvalue.py#L85
2. Deriving scope is expecting a username string at https://github.com/StackStorm/st2/blob/master/st2common/st2common/util/keyvalue.py#L87
3. And RBAC role check requires a UserDB at https://github.com/StackStorm/st2/blob/master/st2common/st2common/util/keyvalue.py#L95